### ISSUE TITLE
gain_phylo crashes

### ISSUE THREAD

**Issue by MyosotisMatt** [Matgend/TDIP#4]
When trying the imputation using `gain_phylo()`, my R session keeps on crashing. 
I have no error to submit.

Here are the packages installed inside my reticulate environment:

 [1] "# packages in environment at C:\\Users\\s1848898\\ANACON~1\\envs\\TDIP:"       
 [2] "#"                                                                             
 [3] "# Name                    Version                   Build  Channel"            
 [4] "absl-py                   2.3.1                    pypi_0    pypi"             
 [5] "astunparse                1.6.3                    pypi_0    pypi"             
 [6] "bzip2                     1.0.8                h0ad9c76_8    conda-forge"      
 [7] "ca-certificates           2025.11.12           h4c7d964_0    conda-forge"      
 [8] "cachetools                6.2.2                    pypi_0    pypi"             
 [9] "certifi                   2025.11.12               pypi_0    pypi"             
[10] "charset-normalizer        3.4.4                    pypi_0    pypi"             
[11] "colorama                  0.4.6                    pypi_0    pypi"             
[12] "flatbuffers               1.12                     pypi_0    pypi"             
[13] "gast                      0.4.0                    pypi_0    pypi"             
[14] "google-auth               2.43.0                   pypi_0    pypi"             
[15] "google-auth-oauthlib      0.4.6                    pypi_0    pypi"             
[16] "google-pasta              0.2.0                    pypi_0    pypi"             
[17] "grpcio                    1.76.0                   pypi_0    pypi"             
[18] "h5py                      3.14.0                   pypi_0    pypi"             
[19] "idna                      3.11                     pypi_0    pypi"             
[20] "importlib-metadata        8.7.0                    pypi_0    pypi"             
[21] "joblib                    1.5.2                    pypi_0    pypi"             
[22] "keras                     2.9.0                    pypi_0    pypi"             
[23] "keras-preprocessing       1.1.2                    pypi_0    pypi"             
[24] "libblas                   3.11.0           2_hf2e6a31_mkl    conda-forge"      
[25] "libcblas                  3.11.0           2_h2a3cdd5_mkl    conda-forge"      
[26] "libclang                  18.1.1                   pypi_0    pypi"             
[27] "libexpat                  2.7.3                hac47afa_0    conda-forge"      
[28] "libffi                    3.5.2                h52bdfb6_0    conda-forge"      
[29] "libhwloc                  2.12.1          default_h64bd3f2_1002    conda-forge"
[30] "libiconv                  1.18                 hc1393d2_2    conda-forge"      
[31] "liblapack                 3.11.0           2_hf9ab0e9_mkl    conda-forge"      
[32] "liblzma                   5.8.1                h2466b09_2    conda-forge"      
[33] "libsqlite                 3.51.0               hf5d6505_0    conda-forge"      
[34] "libwinpthread             12.0.0.r4.gg4f2fc60ca     h57928b3_10    conda-forge"
[35] "libxml2                   2.15.1               h5d26750_0    conda-forge"      
[36] "libxml2-16                2.15.1               h692994f_0    conda-forge"      
[37] "libzlib                   1.3.1                h2466b09_2    conda-forge"      
[38] "llvm-openmp               21.1.6               h4fa8253_0    conda-forge"      
[39] "markdown                  3.9                      pypi_0    pypi"             
[40] "markupsafe                3.0.3                    pypi_0    pypi"             
[41] "mkl                       2025.3.0           hac47afa_454    conda-forge"      
[42] "numpy                     1.22.3                   pypi_0    pypi"             
[43] "oauthlib                  3.3.1                    pypi_0    pypi"             
[44] "openssl                   3.6.0                h725018a_0    conda-forge"      
[45] "opt-einsum                3.4.0                    pypi_0    pypi"             
[46] "packaging                 25.0                     pypi_0    pypi"             
[47] "pip                       25.1.1             pyh8b19718_0    conda-forge"      
[48] "protobuf                  3.19.6                   pypi_0    pypi"             
[49] "pyasn1                    0.6.1                    pypi_0    pypi"             
[50] "pyasn1-modules            0.4.2                    pypi_0    pypi"             
[51] "python                    3.9.23          h8c5b53a_0_cpython    conda-forge"   
[52] "python_abi                3.9                      8_cp39    conda-forge"      
[53] "requests                  2.32.5                   pypi_0    pypi"             
[54] "requests-oauthlib         2.0.0                    pypi_0    pypi"             
[55] "rsa                       4.9.1                    pypi_0    pypi"             
[56] "scikit-learn              1.1.1                    pypi_0    pypi"             
[57] "scipy                     1.13.1                   pypi_0    pypi"             
[58] "setuptools                80.9.0             pyhff2d567_0    conda-forge"      
[59] "six                       1.17.0                   pypi_0    pypi"             
[60] "tbb                       2022.3.0             hd094cb3_1    conda-forge"      
[61] "tensorboard               2.9.1                    pypi_0    pypi"             
[62] "tensorboard-data-server   0.6.1                    pypi_0    pypi"             
[63] "tensorboard-plugin-wit    1.8.1                    pypi_0    pypi"             
[64] "tensorflow                2.9.1                    pypi_0    pypi"             
[65] "tensorflow-estimator      2.9.0                    pypi_0    pypi"             
[66] "tensorflow-io-gcs-filesystem 0.31.0                   pypi_0    pypi"          
[67] "termcolor                 3.1.0                    pypi_0    pypi"             
[68] "threadpoolctl             3.6.0                    pypi_0    pypi"             
[69] "tk                        8.6.13               h2c6b04d_3    conda-forge"      
[70] "tqdm                      4.64.1                   pypi_0    pypi"             
[71] "typing-extensions         4.15.0                   pypi_0    pypi"             
[72] "tzdata                    2025b                h78e105d_0    conda-forge"      
[73] "ucrt                      10.0.26100.0         h57928b3_0    conda-forge"      
[74] "urllib3                   2.5.0                    pypi_0    pypi"             
[75] "vc                        14.3                h2df5915_10  "                   
[76] "vc14_runtime              14.44.35208         h818238b_32    conda-forge"      
[77] "vcomp14                   14.44.35208         h818238b_32    conda-forge"      
[78] "vs2015_runtime            14.44.35208         h38c0c73_32    conda-forge"      
[79] "werkzeug                  3.1.3                    pypi_0    pypi"             
[80] "wheel                     0.45.1             pyhd8ed1ab_1    conda-forge"      
[81] "wrapt                     2.0.1                    pypi_0    pypi"             
[82] "zipp                      3.23.0                   pypi_0    pypi"
